src/transformers/api-history.ts

```diff
@@ -0,0 +1,318 @@
+import logger from '@docusaurus/logger';
+import { Code, Node } from 'mdast';
+import { getJSXImport, isCode, isImport } from '../util/mdx-utils';
+import { ActionTuple, SKIP, visitParents } from 'unist-util-visit-parents';
+import { ElectronVersions, SemVer } from '@electron/fiddle-core';
+import { Parent } from 'unist';
+import { parse as parseYaml } from 'yaml';
+import semver from 'semver';
+import AdmZip from 'adm-zip';
+
+const GH_ACTIONS_ARTIFACTS_URL =
+  'https://api.github.com/repos/electron/website/actions/artifacts';
+
+export type ApiHistory = {
+  added?: { 'pr-url': string }[];
+  deprecated?: { 'pr-url': string; 'breaking-changes-header'?: string }[];
+  changes?: { 'pr-url': string; description: string }[];
+};
+
+export interface PrReleaseVersions {
+  release: string | null;
+  backports: Array<string>;
+}
+export type PrReleaseVersionsContainer = { [key: number]: PrReleaseVersions };
+
+interface PrReleaseArtifact {
+  data: PrReleaseVersionsContainer;
+  endCursor: string;
+}
+
+type GithubArtifactsListResponse = {
+  artifacts: Array<{ id: number; name: string; archive_download_url: string }>;
+};
+
+function isObject(
+  value: unknown
+): value is Record<string | number | symbol, unknown> {
+  return typeof value === 'object' && value !== null && !Array.isArray(value);
+}
+
+function isGithubArtifactsListResponse(
+  value: unknown
+): value is GithubArtifactsListResponse {
+  return isObject(value) && 'artifacts' in value;
+}
+
+function isPrReleaseArtifact(value: unknown): value is PrReleaseArtifact {
+  return isObject(value) && 'data' in value && 'endCursor' in value;
+}
+
+function isApiHistory(value: unknown): value is ApiHistory {
+  return (
+    isObject(value) &&
+    ('added' in value || 'deprecated' in value || 'changes' in value)
+  );
+}
+
+export default function attacher() {
+  return transformer;
+}
+
+function matchApiHistoryCodeBlock(node: Node): node is Code {
+  return (
+    isCode(node) &&
+    node.lang?.toLowerCase() === 'yaml' &&
+    node.meta?.toLowerCase() === 'history'
+  );
+}
+
+let _allElectronVersions: SemVer[];
```

Comment:
```suggestion
let _allElectronVersions: SemVer[] | undefined;
```

More accurate typing.